servant-minimalist.hsfiles

```diff
@@ -0,0 +1,232 @@
+      - tasty-discover
+
+{-# START_FILE Setup.hs #-}
+import           Distribution.Simple
+
+main = defaultMain
+
+{-# START_FILE test/Spec.hs #-}
+{-# OPTIONS_GHC -F -pgmF tasty-discover -optF --tree-display
+ #-}
+-- Driver module responsible for discovering all other modules, and tests inside modules, using tasty-discover preprocessor.
+
+{-# START_FILE test/LibSpec.hs #-}
+module LibSpec where
+
+import           Data.Aeson
+import           Lib
+import           Test.QuickCheck
+import           Test.QuickCheck.Instances ()
+import           Test.Tasty.HUnit
+import           Test.Tasty.QuickCheck
+
+-- Property test
+test_findById_no_user =
+  testProperty "findById for non existing users" $ \(User uid _ _) -> uid /= 1 && uid /= 2 ==> findById uid === Nothing
+
+-- Unit test
+test_findById_users =
+  testCase "findById for existing users" $ do
+    findById (userId albert) @?= Just albert
+    findById (userId isaac) @?= Just isaac
+
+instance Arbitrary User where
+  arbitrary = User <$> arbitrary <*> arbitrary <*> arbitrary
+
+{-# START_FILE src/Lib.hs #-}
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeOperators   #-}
+
+module Lib
+  ( startApp
+  , app
+  , findById
+  , isaac
+  , albert
+  , User(..)
+  ) where
+
+import           Data.Aeson
+import           Data.Aeson.TH
+import           Data.List
+import           Data.Maybe
+import           Network.Wai
+import           Network.Wai.Handler.Warp
+import           Servant
+
+data User = User
+  { userId        :: Int
+  , userFirstName :: String
+  , userLastName  :: String
+  } deriving (Eq, Show)
+
+$(deriveJSON defaultOptions ''User)
+
+-- GET http://localhost:8080/users
+type API
+   = "ping" :> Get '[ JSON] String
+     :<|> "users" :> QueryParam "userId" Int :> Get '[ JSON] [User]
+
+startApp :: IO ()
+startApp = run 8080 app
+
+app :: Application
+app = serve api server
+
+api :: Proxy API
+api = Proxy
+
+server :: Server API
+server = return "pong" :<|> handleUsers
+
+handleUsers :: Maybe Int -> Handler [User]
+handleUsers Nothing    = return users
+handleUsers (Just uid) = return . maybeToList . findById $ uid
+
+isaac = User 1 "Isaac" "Newton"
+
+albert = User 2 "Albert" "Einstein"
+
+users :: [User]
+users = [isaac, albert]
+
+findById :: Int -> Maybe User
+findById requestedId = find (\user -> userId user == requestedId) users
+
+{-# START_FILE app/Main.hs #-}
+module Main where
+
+import           Lib
+
+main :: IO ()
+main = startApp
+
+{-# START_FILE README.md #-}
+# {{name}}
+
+## Build
+`stack build`
+
+## Test
+`stack test`
+
+## Run the server:
+`stack exec {{name}}-exe`
+
+## Try the application
+
+### Ping
+http://localhost:8080/ping
+
+### Users
+http://localhost:8080/users
+http://localhost:8080/users?userId=1
+
+{-# START_FILE ChangeLog.md #-}
+# Changelog for {{name}}
+
+## Unreleased changes
+
+{-# START_FILE .hindent.yaml #-}
+# For Servant API declaration
+line-breaks: [":<|>"]
+
+{-# START_FILE LICENSE #-}
```

Comment:
I personally against suggesting license to users, is there a way for `stack template` to actually ask for a license (like `cabal init`) and generate the file based on it?